code/modules/antagonists/traitor/contractor/syndicate_contract.dm

```diff
@@ -0,0 +1,267 @@
+	payout_type = type
+
+	generate(blacklist)
+
+/datum/syndicate_contract/proc/generate(blacklist)
+	contract.find_target(null, blacklist)
+
+	var/datum/record/crew/record
+	if (contract.target)
+		record = find_record(contract.target.name)
+
+	if (record)
+		target_rank = record.rank
+	else
+		target_rank = "Unknown"
+
+	if (payout_type == CONTRACT_PAYOUT_LARGE)
+		contract.payout_bonus = rand(9,13)
+	else if (payout_type == CONTRACT_PAYOUT_MEDIUM)
+		contract.payout_bonus = rand(6,8)
+	else
+		contract.payout_bonus = rand(2,4)
+
+	contract.payout = rand(0, 2)
+	contract.generate_dropoff()
+
+	ransom = 100 * rand(18, 45)
+
+	var/base = pick_list(WANTED_FILE, "basemessage")
+	var/verb_string = pick_list(WANTED_FILE, "verb")
+	var/noun = pick_list_weighted(WANTED_FILE, "noun")
+	var/location = pick_list_weighted(WANTED_FILE, "location")
+	wanted_message = "[base] [verb_string] [noun] [location]."
+
+/datum/syndicate_contract/proc/handle_extraction(mob/living/user)
+	if (contract.target && contract.dropoff_check(user, contract.target.current))
+
+		var/turf/free_location = find_obstruction_free_location(3, user, contract.dropoff)
+
+		if (free_location)
+			// We've got a valid location, launch.
+			launch_extraction_pod(free_location)
+			return TRUE
+
+	return FALSE
+
+// Launch the pod to collect our victim.
+/datum/syndicate_contract/proc/launch_extraction_pod(turf/empty_pod_turf)
+	var/obj/structure/closet/supplypod/extractionpod/empty_pod = new()
+
+	RegisterSignal(empty_pod, COMSIG_ATOM_ENTERED, PROC_REF(enter_check))
+
+	empty_pod.stay_after_drop = TRUE
+	empty_pod.reversing = TRUE
+	empty_pod.explosionSize = list(0,0,0,1)
+	empty_pod.leavingSound = 'sound/effects/podwoosh.ogg'
+
+	new /obj/effect/pod_landingzone(empty_pod_turf, empty_pod)
+
+/datum/syndicate_contract/proc/enter_check(datum/source, sent_mob)
+	SIGNAL_HANDLER
+
+	if(!istype(source, /obj/structure/closet/supplypod/extractionpod))
+		return
+	if(!isliving(sent_mob))
+		return
+	var/mob/living/person_sent = sent_mob
+	var/datum/antagonist/traitor/traitor_data = contract.owner.has_antag_datum(/datum/antagonist/traitor)
+	if(person_sent == contract.target.current)
+		traitor_data.uplink_handler.contractor_hub.contract_TC_to_redeem += contract.payout
+		traitor_data.uplink_handler.contractor_hub.contracts_completed++
+		if(person_sent.stat != DEAD)
+			traitor_data.uplink_handler.contractor_hub.contract_TC_to_redeem += contract.payout_bonus
+		status = CONTRACT_STATUS_COMPLETE
+		if(traitor_data.uplink_handler.contractor_hub.current_contract == src)
+			traitor_data.uplink_handler.contractor_hub.current_contract = null
+	else
+		status = CONTRACT_STATUS_ABORTED // Sending a target that wasn't even yours is as good as just aborting i
+		if(traitor_data.uplink_handler.contractor_hub.current_contract == src)
+			traitor_data.uplink_handler.contractor_hub.current_contract = null
+
+	if(iscarbon(person_sent))
+		for(var/obj/item/person_contents in person_sent.contents)
```

Comment:
We refactored this in kidnapping to avoid it from doing shit like pulling out all of a person's implants, might be worth comparing with this proc.

Reply:
i thought it was actually intentional to not bring implants over lol